schematic/store/synapse.py

```diff
@@ -705,7 +705,10 @@ def getFilesInStorageDataset(
             ValueError: Dataset ID not found.
         """
         file_list = []
-
+        # HACK: must requery the fileview to get new files, since SynapseStorage will query the last state
+        # of the fileview which may not contain any new folders in the fileview.
+        # This is a workaround to fileviews not always containing the latest information
+        self.query_fileview(force_requery=True)
```

Comment:
Post in the team channel to discuss this hack and there's a message in #synapse channel for us to chime in on

Reply:
I wonder if there is something that we can do here with this API:

https://rest-docs.synapse.org/rest/POST/entity/id/table/query/async/start.html

Specifically:
"The last updated on date of the table (lastUpdatedOn) = 0x80"

If this get's updated as data is indexed into the table we could fetch the lastUpdatedOn field before we query the table to know if we need to re-query the table or not.

Reply:
Another thing to note: if we look at the failing test that was the motivation behind this hack, it's because it falls outside of the overall schematic flow. 

So for example: when people upload a bunch of files, they usually first run manifest generate. Now manifest generate will have a "this dataset doesn't exist" situation but then afterwards, the fileview should always consist of the data - theoretically. There have been incidents when it doesn't but that's very little in the grand scheme of things.

So we _could_ remove the hack and trigger the fileview indexing within the test but that's probably best as a team decision.

Reply:
@GiaJordan After a team discussion, we decided that it would be better to modify the test to re-query the fileview since this is such an edgecase.  It does throw an error already, and upon multiple runs of this code outside of the test, it would work.  This only doesn't work so smoothly in the test because resources are created and destroyed.

```
FAILED tests/integration/test_metadata_model.py::TestMetadataModel::test_submit_filebased_manifest_file_and_entities_valid_manifest_submitted - LookupError: Dataset syn64313762 could not be found in fileview syn23643253.
FAILED tests/integration/test_metadata_model.py::TestMetadataModel::test_submit_filebased_manifest_file_and_entities_mock_filename - LookupError: Dataset syn64313765 could not be found in fileview syn23643253.
```